File: test2.py
import logging
logging.basicConfig(level=logging.DEBUG)

import tracing
import time

# tracer = tracing.Tracer.withURLsAndToken("ws://localhost:52690/ws", None, "fakeToken")
tracer = tracing.Tracer.withURLsAndToken("wss://tracing-develop.datawire.io/ws", None, None)

def goodHandler(result):
	print("Good!")

	for record in result:
		print(record.toString())
		
def badHandler(result):
	print("Failure: %s" % result.toString())

def formatLogRecord(record):
	return("%.3f %s %s" % (record.timestamp, record.record.level, record.record.text))

tracer.poll().andEither(goodHandler, badHandler)

File: test2.js
var tracing = require("datawire_tracing").tracing;

tracer = new tracing.Tracer()

tracer.query(1467045724000, 1467045780000).andEither(
	function goodHandler(res) {
		console.log("Good!");

		res.result.forEach(function (record) {
			console.log(String(record.timestamp) + " " + record.record.level +
				        " " + record.record.text);
		});

		# This should be unnecessary -- to be fixed soon.
		return null;
	},
	function badHandler(result) {
		print("Failure: %s" % result.toString());

		# This should be unnecessary -- to be fixed soon.
		return null;
	}
);
